### PR TITLE
snapshot ecosystem, subgraph methods from fee allocator

### PR DESCRIPTION
**Pull request by jalbrekt85**
#3 #4

### CODE REVIEW

File: bal_tools/subgraph.py [hunk 1]
```diff
@@ -96,3 +102,136 @@ def get_first_block_after_utc_timestamp(self, timestamp: int) -> int:
...
+    def fetch_all_pools_info(self) -> List[Pool]:
+        """
+        Fetches all pools info from balancer graphql api
+        """
+        res = self.fetch_graphql_data(
+            "core", "vebal_get_voting_list", url="https://api-v3.balancer.fi"
+        )
+        return [Pool(**pool) for pool in res["veBalGetVotingList"]]
+
+    def get_balancer_pool_snapshots(
```
Comment (jalbrekt85) [inline bal_tools/subgraph.py:218]: still using the traditional endpoints for this, see: https://github.com/BalancerMaxis/bal_tools/issues/4#issuecomment-2113277322

**Review by Tritium-VLK — APPROVED** [review]

File: bal_tools/ecosystem.py [hunk 2]
```diff
@@ -115,3 +120,82 @@ def get_aura_pid_from_gauge(self, deposit_gauge_address: str) -> int:
                     f"Got a list result with something other than 1 member when compiling aura PID mapping: {result}"
                 )
             return result[0]
+
```
Comment (Tritium-VLK) [inline bal_tools/ecosystem.py:123]: I did not review this file in detail but it is not in operation anywhere yet so we will see how it works

File: bal_tools/graphql/blocks/first_block_after_ts.gql [hunk 3]
```diff
@@ -1,9 +1,9 @@
-query FirstBlockAfterTimestamp($timestamp: BigInt) {
+query FirstBlockAfterTimestamp($timestamp_gt: BigInt, $timestamp_lt: BigInt) {
   blocks(
     first: 1
     orderBy: number
     orderDirection: asc
-    where: { timestamp_gt: $timestamp }
+    where: { timestamp_gt: $timestamp_gt, timestamp_lt: $timestamp_lt}
```
Comment (Tritium-VLK) [inline bal_tools/graphql/blocks/first_block_after_ts.gql:6]: Are you sure nothing is using this yet and passing only GT?

File: bal_tools/requirements.txt [hunk 4]
```diff
@@ -5,5 +5,6 @@ web3
 dotmap
 munch==4.0.0
 gql[requests]
+dacite
 
 git+https://github.com/BalancerMaxis/bal_addresses@0.9.1
```
Comment (Tritium-VLK) [inline bal_tools/requirements.txt:10]: bump this?

File: bal_tools/subgraph.py [hunk 5]
```diff
@@ -37,16 +45,21 @@ def get_subgraph_url(self, subgraph="core") -> str:
...
             magic_word = "subgraph:"
         elif subgraph == "gauges":
             magic_word = "gauge:"
         elif subgraph == "blocks":
             magic_word = "blocks:"
-            ## UI has no blocks subgraph for op
+            ## UI has no blocks subgraph for op or gnosis
             if chain == "optimism":
                 return "https://api.thegraph.com/subgraphs/name/iliaazhel/optimism-blocklytics"
+            elif chain == "gnosis-chain":
```
Comment (Tritium-VLK) [inline bal_tools/subgraph.py:61]: A little unclear why you moved it and unsure of this ordering, can you explain?

**Review by Tritium-VLK — CHANGES_REQUESTED** [review]
Loooks pretty good to me.  Before I just stamp it.  Posted a few comments. 

I didn't do a lot of review of new functions/code/gql files, nor tests. I assume that we can hash this out once we start using it a bit, and we are still in early versions of bal_tools.  What's important is that we don't break anything, or notice/change it fast if/when we do.

File: bal_tools/requirements.txt (hunk 4 above)
Comment (jalbrekt85) [reply bal_tools/requirements.txt:11]: migrated to pydantic: c30415a

File: bal_tools/subgraph.py (hunk 5 above)
Comment (jalbrekt85) [reply bal_tools/subgraph.py:62]: wdym ordering? i dont think it should matter here

File: bal_tools/graphql/blocks/first_block_after_ts.gql (hunk 3 above)
Comment (jalbrekt85) [reply bal_tools/graphql/blocks/first_block_after_ts.gql:6]: not sure where else its used outside of this/fee allocator currently. this query is called from `get_first_block_after_utc_timestamp`, so should be implied that its gt

**Pull request merged**